File: poulet-mecanique-frontend/src/app/data/discord-channel.ts
export class DiscordChannel {

    id: string;
    name: string;
    category: string;
    label: string;

    constructor(id: string, name: string, category: string, label: string) {
        this.id = id;
        this.name = name;
        this.category = category;
        this.label = label;
    }
}

File: poulet-mecanique-backend/src/main/java/fr/arkyan/popo/pouletmecaniquebackend/data/dto/DiscordChannel.java
package fr.arkyan.popo.pouletmecaniquebackend.data.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class DiscordChannel {

    private String id;
    private String name;
    private String category;

    @JsonProperty("label")
    public String getLabel() {
        return category + " - " + name;
    }

}
